src/theory/arith/nl/transcendental/transcendental_state.h

```diff
@@ -181,18 +191,21 @@ struct TranscendentalState : protected EnvObj
   /**
    * Some transcendental functions f(t) are "purified", e.g. we add
    * t = y ^ f(t) = f(y) where y is a fresh variable. Those that are not
-   * purified we call "master terms".
+   * purified we call "purified terms".
    *
-   * The maps below maintain a master/slave relationship over
-   * transcendental functions (SINE, EXPONENTIAL, PI), where above
-   * f(y) is the master of itself and of f(t).
+   * The maps below maps transcendental function applications (SINE,
+   * EXPONENTIAL, PI) to their purified version, where above
+   * f(y) is the purified version of itself and of f(t).
    *
    * This is used for ensuring that the argument y of SINE we process is on
    * the interval [-pi .. pi], and that exponentials are not applied to
    * arguments that contain transcendental functions.
    */
-  std::map<Node, Node> d_trMaster;
-  std::map<Node, std::unordered_set<Node>> d_trSlaves;
+  NodeMap d_trPurify;
+  /** inverse mapping of above */
+  NodeMap d_trPurifies;
```

Comment:
A general question about these that is not totally clear to me from the code: Do these maps (conceptually) have a one-to-one relationship? In particular: can it every happen that two nodes are purified to the same purification term? (I don't think so, but not entirely sure).
If this is the case, please make this explicit by asserting that the purification term is indeed new when inserting stuff in `initLastCall()`.